lib/puppet.rb

```diff
@@ -36,6 +50,29 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    if GETTEXT_AVAILABLE
+      # TODO: Handle the rubygems case?
+      local_locale_path = File.absolute_path('../locales', File.dirname(__FILE__))
+      posix_system_locale_path = File.absolute_path('../../../share/puppet/locale', File.dirname(__FILE__))
+      win32_system_locale_path = File.absolute_path('../../../../../puppet/share/puppet/locale', File.dirname(__FILE__))
+
+      if File.exist?(local_locale_path)
+        locale_path = local_locale_path
+      elsif File.exist?(win32_system_locale_path)
+        locale_path = win32_system_locale_path
+      elsif File.exist?(posix_system_locale_path)
+        locale_path = posix_system_locale_path
+      else
+        # We couldn't load our locale data. Possibly we're loaded as a rubygem or something?
+        locale_path = nil
+      end
+
+      if locale_path
+        GettextSetup.initialize(locale_path)
+        FastGettext.locale = GettextSetup.negotiate_locale(ENV["LANG"])
```

Comment:
The gem does appear to work on Windows.  It's using the older API call GetThreadLocaleID whereas the more modern call should be GetUserDefaultLocaleName. https://msdn.microsoft.com/en-us/library/windows/desktop/dd318136(v=vs.85).aspx.  To be honest though, we could probably implement the call ourselves instead of pulling in yet-another-gem.